src/Microsoft.OData.Core/ODataReaderCore.cs

```diff
@@ -853,22 +867,9 @@ protected internal class Scope
             /// </summary>
             /// <param name="state">The reader state of this scope.</param>
             /// <param name="item">The item attached to this scope.</param>
-            /// <param name="navigationSource">The navigation source we are going to read entities for.</param>
-            /// <param name="expectedResourceType">The expected resource type for the scope.</param>
             /// <param name="odataUri">The odataUri parsed based on the context uri for current scope</param>
-            /// <remarks>The <paramref name="expectedResourceType"/> has the following meanings for given state:
-            /// Start -               it's the expected base type of the top-level resource or resources in the top-level resource set.
-            /// ResourceSetStart -           it's the expected base type of the resources in the resource set.
-            ///                       note that it might be a more derived type than the base type of the entity set for the resource set.
-            /// EntryStart -          it's the expected base type of the resource. If the resource has no type name specified
-            ///                       this type will be assumed. Otherwise the specified type name must be
-            ///                       the expected type or a more derived type.
-            /// NestedResourceInfoStart - it's the expected base type the entries in the expanded link (either the single resource
-            ///                       or entries in the expanded resource set).
-            /// EntityReferenceLink - it's null, no need for types on entity reference links.
-            /// In all cases the specified type must be an structured type.</remarks>
             [SuppressMessage("Microsoft.Performance", "CA1800:DoNotCastUnnecessarily", Justification = "Debug.Assert check only.")]
-            internal Scope(ODataReaderState state, ODataItem item, IEdmNavigationSource navigationSource, IEdmStructuredType expectedResourceType, ODataUri odataUri)
+            internal Scope(ODataReaderState state, ODataItem item, ODataUri odataUri)

```

Comment:
>Scope [](start = 21, length = 5)

do we have the out call for this constructor? If no, i don't think we should create a one like this.

Reply:
This is called in a couple of other places.

---
In reply to: [266674779](https://github.com/OData/odata.net/pull/1322#discussion_r266674779) [](ancestors = 266674779)